pkg/controller/jobframework/reconciler.go

```diff
@@ -745,17 +745,17 @@ func equivalentToWorkload(ctx context.Context, c client.Client, job GenericJob,
 	jobPodSets := clearMinCountsIfFeatureDisabled(job.PodSets())
 
 	if runningPodSets := expectedRunningPodSets(ctx, c, wl); runningPodSets != nil {
-		if equality.ComparePodSetSlices(jobPodSets, runningPodSets) {
+		if equality.ComparePodSetSlices(jobPodSets, runningPodSets, workload.IsAdmitted(wl)) {
```

Comment:
This ignoring of tolerations is done independently of the framework, but there is no issue with jobs, just with pods. I'm wondering if we should make the fix more specific. WDYT @alculquicondor ?

Reply:
I'm ok with this for now. But can we use the condition status of QuotaReserved, instead of Admitted?

Reply:
I don't think it makes any difference, the toleration are only changing after the pods are unsuspended.

Reply:
sg, thanks for the reminder.